controllers/drplacementcontrol.go

```diff
@@ -1135,6 +1348,48 @@ func (d *DRPCInstance) EnsureCleanup(clusterToSkip string) error {
 		return nil
 	}
 
+	d.log.Info(fmt.Sprintf("PeerReady Condition %v", condition))
+
+	err := d.moveVRGToSecondaryOnPeers(clusterToSkip)
```

Comment:
If VRG MW was deleted on the managed cluster in a previous reconcile, but VRG was not yet deleted on the cluster (IOW MCV returns a VRG), then the next reconcile would attempt to move VRG to Secondary here, and fail to find a VRG. Further reconciles will keep looping here.

It looks like VRG is being moved to secondary so that if it is retained for VolSync it can act on the Secondary case. The call should possibly be moved into the VolSync handling if condition below.

Reply:
Right. That's a bug.  Fixed in the latest commit: [4d9ec71a72a815c2084564a037fba89c47a1c133](https://github.com/RamenDR/ramen/pull/434/commits/4d9ec71a72a815c2084564a037fba89c47a1c133)